numpy/linalg/linalg.py

```diff
@@ -1749,7 +1750,8 @@ def det(a):
     _assertNdSquareness(a)
     t, result_t = _commonType(a)
     signature = 'D->D' if isComplexType(t) else 'd->d'
-    return _umath_linalg.det(a, signature=signature).astype(result_t)
+    r = _umath_linalg.det(a, signature=signature)
+    return r.astype(result_t)
```

Comment:
Ah, I see... There also seem to be tests that would catch if someone tried to introduce that optimization in the future, so I guess it is OK.

Perhaps it is worth a comment? What about branching on `np.isscalar(r)`?